// scratch-vm/src/extensions/scratch3_testExtension/index.js
//import {Pdollarbox} from '../../../../../src/components/pdollarbox/pdollarbox.jsx';

class GestureBlock {
    getInfo() {
        return {
            "id": "GestureBlock",
            "name": "GestureBlock",
            "blocks": [{
                    "opcode": "gestureCheck",
                    "blockType": "hat",
                    "text": "When [gestureID] is recognized",
                    "arguments": {
                        "gestureID": {
                            "type": "number",
                            "defaultValue": "T",
                           "menu": 'KEY'
                        }  
                       
                    }
                
            }],
            "menus": { //we will get back to this in a later tutorial
                 KEY: {
                    acceptReporters: true,
                    items: [
                        
                        {text: 'T', value: 'T'},
                        {text: 'N', value: 'N'},
                        {text: 'P', value: 'P'},
                        {text: 'D', value: 'D'},
                        {text: 'X', value: 'X'},
                        {text: 'H', value: 'H'},
                        {text: 'I', value: 'I'}
                    ]
                }
            }
        };
    }
    gestureCheck({gestureID}) {

        

        if(gestureID == localStorage.getItem('mostRecentGesture'))
        {
            localStorage.setItem('mostRecentGesture', 'z');
            return true;
        }
        else
        {
            return false;
        }

    };
}

module.exports =  GestureBlock;
